File: c++/ProxyPattern/Proxy.cpp
#include "stdafx.h"
#include "Proxy.h"


Proxy::Proxy(Google google)
{
	this->google = google;
}

void Proxy::Request(string url)
{
	this->google.Request(url);
}


string Proxy::Response()
{
	return this->google.Response();
}

File: csharp/ProxyPattern/Proxy.cs

namespace ProxyPattern
{
    public class Proxy : WebSite
    {
        private Google google;

        public Proxy(Google google)
        {
            this.google = google;
        }

        public override void Request(string url)
        {
            this.google.Request(url);
        }

        public override string Response()
        {
            return this.google.Response();
        }
    }
}
